src/main/java/Duke.java

```diff
@@ -1,10 +1,148 @@
+                continue;
+            }
+            if(commandEntered.equals("list")) {        //command list lists out all recorded words
+                listOfTasksPrinter(orderAdded, recordedTask);
+                continue;
+            }
+            if(commandEntered.contains("done ")) {      //command done<space>(number) changes task status of a task from not done to done i.e cross to tick
+                setTaskAsDone(commandEntered, recordedTask);
+                continue;
+            }
+            if(commandEntered.equals("bye")) { //command bye breaks the loop and ends the interaction with Dukebot
+                break;
+            }
+            recordedTask[orderAdded-1] = new Task(orderAdded, commandEntered, false);
+            orderAdded = genericTaskAdder(orderAdded, recordedTask);
+        }
+        printByeMessage(); //message to bid farewell to users
+    }
+
+    private static int specificTaskAdder(int orderAdded, Task[] recordedTask, String commandEntered) {
+        String[] taskCommandArr = commandEntered.split(" ", 2);
+        String exactDueDate = "";
+        if(taskCommandArr[1].contains("/by")) {
+            String taskDueDateString = taskCommandArr[1];
+            String[] taskDueDateArr = taskDueDateString.split("/by", 2);
+            exactDueDate = "(by:" + taskDueDateArr[1] + ")";
+            taskCommandArr[1] = taskDueDateArr[0];
+        }
+        else if(taskCommandArr[1].contains("/at")) {
+            String taskDueDateString = taskCommandArr[1];
+            String[] taskDueDateArr = taskDueDateString.split("/at", 2);
+            exactDueDate = "(at:" + taskDueDateArr[1] + ")";
+            taskCommandArr[1] = taskDueDateArr[0];
+        }
+        switch(taskCommandArr[0]) {
+        case "todo":
+            recordedTask[orderAdded -1] = new ToDos(orderAdded, taskCommandArr[1], false);
+            recordedTask[orderAdded -1].setTaskType("T");
+            break;
+        case "deadline":
+            recordedTask[orderAdded -1] = new Deadlines(orderAdded, taskCommandArr[1], false);
+            recordedTask[orderAdded -1].setTaskType("D");
+            break;
+        case "event":
+            recordedTask[orderAdded -1] = new Events(orderAdded, taskCommandArr[1], false);
+            recordedTask[orderAdded -1].setTaskType("E");
+            break;
+        default:
+            recordedTask[orderAdded-1] = new Task(orderAdded, commandEntered, false);
+            orderAdded = genericTaskAdder(orderAdded, recordedTask);
+            return orderAdded;
+           // break;
+        }
+        recordedTask[orderAdded-1].setTaskName(taskCommandArr[1] + exactDueDate);
+
+        System.out.println(LINE_DIVIDER);
+        System.out.println("Got it. I've added this task: ");
+        System.out.println("  [" + recordedTask[orderAdded -1].getCurrentTaskType() + "][" + recordedTask[orderAdded -1].taskStatus() + "] " + recordedTask[orderAdded-1].getTaskName());
```

Comment:
Maybe you can shorten this line? The hard character limit is 120 characters. It seems like it occurs other times too.

Reply:
Cleaned it up, thank you!